### PR TITLE
Fix initial project setup

### PR DESCRIPTION
**Pull request by danielknell**
- cleared out unused code from bootstrap.sh
- added npm_install to Makefile in line with other projects
- added npm_install to run_all directive in Makefile
- made run_app use the virtualenv python
- used DM_ENVIRONMENT to infer a GULP_ENVIRONMENT in order to build development assets on development environments and production assets in all other environments

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,4 +43,4 @@ show_environment:
 	@echo "Environment variables in use:"
 	@env | grep DM_ || true
 
-.PHONY: run_all run_app virtualenv requirements requirements_for_test frontend_build test test_pep8 test_python test_javascript show_environment
+.PHONY: run_all run_app virtualenv requirements requirements_for_test npm_build frontend_build test test_pep8 test_python test_javascript show_environment
```
Comment (idavidmcdonald) [inline Makefile]: Sorry, my knowledge is not great about make files. Where is this `npm_build` defined? I see `npm_install`, although that is not listed in the `.PHONY`, how come?

Comment (idavidmcdonald) [inline Makefile]: My guess is that `npm_build` should instead be `npm_install`?

Comment (danielknell) [inline Makefile]: fixed

**Comment by idavidmcdonald** [comment]
👍 Looks good when the tests are green.

**Pull request merged**